space.go

```diff
@@ -782,17 +782,11 @@ func (space *Space) Step(dt float64) {
 }
 
 func (space *Space) Lock() {
-	space.locked++
```

Comment:
This would be a behavioral change. With the int someone could be doing:

```
space.Lock()
space.Lock()
space.Unlock()
space.Unlock()
```

By changing to a boolean this will now error. 

I'm not sure why this would matter but I want to keep close to the original source as possible.